hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestBlockStoragePolicy.java

```diff
@@ -1291,6 +1291,9 @@ public void testChooseTargetWithTopology() throws Exception {
         new HashSet<Node>(), 0, policy2, null);
     System.out.println(Arrays.asList(targets));
     Assert.assertEquals(3, targets.length);
+    if (namenode != null) {
+      namenode.stop();
+    }
```

Comment:
This should be in a finally block. eg:
Namenode namenode = new Namenode(conf)
 try {
     Do Something
} finally {
  if (namenode != null) {
      namenode.stop();
}

The reason being, in case the test fails after creation of namenode, in that case if it isn't in the finally block the stop command won't be executed.

Reply:
Thanks @ayushtkn  for your review,  have just updated the code